src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -37,7 +37,7 @@ public function validate($value, Constraint $constraint)
         $valid = filter_var($value, FILTER_VALIDATE_EMAIL);
 
         if ($valid) {
-            $host = substr($value, strpos($value, '@') + 1);
+            $host = $this->extractHostname($value);
```

Comment:
@xabbuh  @nicolas-grekas thanks for the feedback!  Can you see my comment in the `EmailValidatorTest` class, and perhaps give a pointer or two on exactly how this change can be accurately tested via the public API?